### ISSUE TITLE
Generator doesn't work with Node 10

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster#7551]
I know we only support Node.js LTS, but we might want to look into supporting Node 10. Currently, the following error happens when you try to use Node 10.

```
WARNING! Something went wrong fetching the latest JHipster version number...
(node:15239) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability
issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

**Comment by jdubois** [comment]
Yes I've been playing with it in the last few days, so let me give some insights:

- You will always get the warning, and that's not due to JHipster, so we can ignore it for the moment (they'll fix it on a later Node 10 release, hopefully).
- You can make it work (I had it working), but it's very falky when you do `yarn link`. It was broken on my machine, then it worked for a few hours, and now it looks broken beyond repair. I guess this is linked with the new NPM 6 release. Then, for normal users (who won't do `yarn link`), that might not a be an issue at all.

So for me this is unstable, as always if you are on a non-LTS release (early Node 9 releases were awful), and we should wait a little before using it, but it's not that bad as I could work with it for several hours without much trouble (excepted the warning above).

Now my issue is that this version comes automatically with Homebrew on Mac, which means people doing `brew install jhipster` on Mac will have a broken release. I don't think we can correct this, as with Homebrew there is a dependency between Yarn and the latest Yarn release....

We would have less trouble by migrating back to NPM and leaving Yarn, maybe it's time to open up this discussions again @pascalgrimaud ?

On a personal note I'm switching back to Node 8, and installing it manually (no Homebrew).

**Comment by pascalgrimaud** [comment]
I'm not against switch back to NPM. It's the same for me. If it's more consistent with release and easier for npm link, so let's switch back.
But I'm not sure everybody in the core team will agree about that.

Here all solutions:
- stay like this, support both, Yarn by default, NPM with flag
- support both, NPM by default, Yarn with flag
- switch to NPM and remove completely Yarn

So let discuss. 
Once it's decided, I can do it, I know what to do (generator, Docker, Devbox, Zsh, documentation, etc...)

**Comment by PierreBesson** [comment]
I'm not sure what this has to do with the npm/yarn debate as this is about node itself. I'm personally not OK with dropping yarn as the transitive dependency fixing is better than with npm.
The real issue here is that the nodejs "current" version is not always backward compatible and people are always super quick to jump on it. Strangely they are not that eager to upgrade their java version :stuck_out_tongue_winking_eye:...
IMO we should just make super clear that we only support the lts version of node. If needed I could code a quick check code to warn users using unsupported node versions.

**Comment by PierreBesson** [comment]
@jdubois, it seems that you can install node 8 on a Mac with `brew install node@8`. I've not tried it yet but I will as soon as I unwrap my new work macbook.

**Comment by jdubois** [comment]
Yes that will install Node 8, but when you will install Yarn it will transitively pull Node 10...
It's much easier to install it manually

**Comment by PierreBesson** [comment]
@jdubois I just managed to install yarn and nodejs8 with brew with the following commands:

> brew install node@8
> brew link --force node@8
> brew install yarn --without-node

**Comment by mraible** [comment]
The error I noted in this issue's description actually happens now, so that's not the cause, the warning before is.

```
Deploying application
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Node.js app detected
remote:
remote: -----> Creating runtime environment
remote:
remote:        NPM_CONFIG_LOGLEVEL=error
remote:        NPM_CONFIG_PRODUCTION=false
remote:        NODE_VERBOSE=false
remote:        NODE_ENV=production
remote:        NODE_MODULES_CACHE=true
remote:
remote: -----> Installing binaries
remote:        engines.node (package.json):  >=6.9.0
remote:        engines.npm (package.json):   unspecified (use default)
remote:        engines.yarn (package.json):  unspecified (use default)
remote:
remote:        Resolving node version >=6.9.0...
remote:        Downloading and installing node 10.0.0...
remote:        Using default npm version: 5.6.0
remote:        Resolving yarn version 1.x...
remote:        Downloading and installing yarn (1.6.0)...
remote:        Installed yarn 1.6.0
remote:
remote: -----> Restoring cache
remote:        Skipping cache restore (not-found)
remote:
remote: -----> Building dependencies
remote:        Installing node modules (yarn.lock)
remote:        yarn install v1.6.0
remote:        (node:230) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
remote:        [1/4] Resolving packages...
```

**Comment by pascalgrimaud** [comment]
I tested Node10 in our Hipster-Labs, here all the builds:
https://travis-ci.org/hipster-labs/jhipster-travis-build/builds/378394590

So I confirm, with Node10, it failed for 3 configs, with the same error mentionned by Matt

**Comment by PierreBesson** [comment]
@pascalgrimaud, It looks like this is a Yarn bug that got fixed just today !
See: https://github.com/yarnpkg/yarn/issues/5477
And the fix: https://github.com/yarnpkg/yarn/commit/546a1576edbf701021ce65e3dd8daff267083f52
It seems the issue lies with V8-compile-cache that an API deprecated in Node 10 https://github.com/zertosh/v8-compile-cache/commit/f6f5dfa217c55ddda5e950a08751d943d4cb2833

**Comment by pascalgrimaud** [comment]
Oh ok, let's wait the next release of Yarn

**Comment by PierreBesson** [comment]
OK so this is going to get fixed in the next version of yarn, it displays a lot of warning every time you run yarn but is only a blocker for `yarn e2e`.

However the issue from stackoverflow is a different one. It is caused by node-saas trying to recompile itself. And indeed this does not work in Heroku as you need a g++ toolchain.

**Comment by pascalgrimaud** [comment]
So after our discussion, at Ippon, I think we agree on:
- keeping Yarn as default and still support both: Yarn and NPM (stay like today)
- supporting Node LTS, and have an eye on Node 10, I just added a daily build for this [node-latest branch](https://github.com/hipster-labs/jhipster-travis-build/tree/node-latest)

**Comment by deepu105** [comment]
Thats great Pascal

Thanks & Regards,
Deepu

**Comment by PierreBesson** [comment]
Yarn just released V1.7.0 which add compayibility with Node 10 : https://twitter.com/yarnpkg/status/999404494906712066?s=19
